### PR TITLE
Fix multiple email notifications for clinical biospecimen survey

### PR DESCRIPTION
**Pull request by we-ai**
This PR fixes issue #608(https://github.com/episphere/connect/issues/608). Below are the details:
- Fixed issue #608: possible of sending multiple notification emails for clinical biospecimen survey.
- Introduced a minimal state manager for easier track of the state data of the app.
- Removed some code that are either commented out or logically not used.
- Added documentation to functions for clarity.
- Renamed variable for clarity: getVerifiedParticipants -> `verifiedParticipants`
- Renamed functions for clarity: getWorflow -> `getWorkflow`, proccedToSpecimenPage -> `proceedToSpecimenPage`

### CODE REVIEW

File: src/pages/dashboard.js [hunk 1]
```diff
@@ -170,26 +169,31 @@ export const searchResults = (result) => {
...
+
     result.forEach(data => {
 
         if(data['821247024'] === 922622075) return;
+
+        let tdTemplate=`
+            <td>${data['996038075']}</td>
+            <td>${data['399159511']}</td>
+            <td>${data['231676651']}</td>
+            <td>${data['564964481']}/${data['795827569']}/${data['544150384']}</td>
```
Comment (anthonypetersen) [inline src/pages/dashboard.js:186]: THANK YOU FOR DOING THIS!!!

File: src/events.js [hunk 2]
```diff
@@ -2043,18 +2037,10 @@ const btnsClicked = async (connectId, formData) => {
...
     formData['820476880'] = collectionID;
-    formData['650516960'] = getWorflow() === 'research' ? 534621077 : 664882224;
+    formData['650516960'] = getWorkflow() === 'research' ? 534621077 : 664882224;
     formData['Connect_ID'] = parseInt(document.getElementById('specimenLinkForm').dataset.connectId);
     formData['token'] = document.getElementById('specimenLinkForm').dataset.participantToken;
     
-    let bloodAccessionId = formData?.['646899796'];
-    if (bloodAccessionId) {
-        formData['646899796'] = bloodAccessionId;
-    }
```
Comment (anthonypetersen) [inline src/events.js:2053]: Where did this logic go?

File: src/shared.js [hunk 3]
```diff
@@ -253,9 +274,12 @@ export const performSearch = async (query) => {
     showAnimation();
     const response = await findParticipant(query);
     hideAnimation();
-    const getVerifiedParticipants = response.data.filter(dt => dt['821247024'] === 197316935);
-    if(response.code === 200 && getVerifiedParticipants.length > 0) searchResults(getVerifiedParticipants);
-    else if(response.code === 200 && getVerifiedParticipants.length === 0) showNotifications({title: 'Not found', body: 'The participant with entered search criteria not found!'}, true)
+    const verifiedParticipants = response.data.filter(dt => dt['821247024'] === 197316935);
+
+    if (response.code === 200 && verifiedParticipants.length > 0) {searchResults(verifiedParticipants);
```
Comment (anthonypetersen) [inline src/shared.js:279]: this is so small... but can you structure the code within the brackets for the IF statement similar to what you did with the ELSE IF statement? (code is in line between the brackets)

File: .gitignore [hunk 4]
```diff
@@ -3,4 +3,5 @@
 static/.DS_Store
 static/images/.DS_Store
 scratch.js
-/tmp
\ No newline at end of file
+/tmp
```
Comment (anthonypetersen) [inline .gitignore:6]: what is .hintrc?

File: src/events.js (hunk 2 above)
Comment (we-ai) [reply src/events.js:2053]: It goes nowhere. It's like `if (a) { a=a; }`. This is why I removed it.

File: index.html [hunk 5]
```diff
@@ -16,7 +16,7 @@
 </head>
 <body>
     <nav class="navbar navbar-expand-lg navbar-light bg-light fixed-top">
-        <a class="navbar-brand" href="#"><img height="45px" width="140px" src="./static/images/NCI-LOGO-FA-Color-Blue.png"></a> 
+        <a class="navbar-brand" href="#"><img height="45px" width="140px" alt="Connect blue logo" src="./static/images/NCI-LOGO-FA-Color-Blue.png"></a>
```
Comment (anthonypetersen) [inline index.html:19]: was this just preference at add an "ALT" tag to this? I can't see where we are referencing this

File: src/events.js (hunk 2 above)
Comment (anthonypetersen) [reply src/events.js:2053]: Oops, I see that now... I misread what this was doing.. Thanks!

File: index.html (hunk 5 above)
Comment (we-ai) [reply index.html:19]: It's not referred. Adding it is to follow the rules about `img` element.

File: src/shared.js (hunk 3 above)
Comment (we-ai) [reply src/shared.js:279]: Make it beautiful. Sure.

File: index.html (hunk 5 above)
Comment (anthonypetersen) [reply index.html:19]: oh, is that just an industry standard?

File: src/shared.js (hunk 3 above)
Comment (anthonypetersen) [reply src/shared.js]: thank you 🥇

File: src/events.js [hunk 6]
```diff
@@ -1753,33 +1753,21 @@ export const addGoToCheckInEvent = () => {
 };
 
 export const addGoToSpecimenLinkEvent = () => {
-    // For research
-    const researchSpecimenLinkButtons = document.querySelectorAll('[data-specimen-link-connect-id]');
-    Array.from(researchSpecimenLinkButtons).forEach((btn) => {
-        btn.addEventListener('click', async () => {
+    const workflow = getWorkflow();
```
Comment (anthonypetersen) [inline src/events.js:1756]: these lines referencing workflow could probably be removed? workflow can only be "research" or "clinical", so doesn't seem like there is a scenario we are trying to keep out with this IF statement

File: src/events.js [hunk 7]
```diff
@@ -2294,6 +2285,8 @@ const proccedToSpecimenPage = async (accessionID1, accessionID3, selectedVisit,
             const yesBtn = document.getElementById('addCollection');
             yesBtn.addEventListener("click", async e => {
                 formData.collectionId = bloodAccessionId?.data?.[820476880];
+                formData['331584571'] =  selectedVisit;
```
Comment (anthonypetersen) [inline src/events.js:2288]: was this CID not getting set properly in this scenario?

Comment (we-ai) [reply src/events.js:2288]: True. It is not set correctly. This is  1 of the 2 lines needed for the notification fix.

**Review by abhinavjonnada82 — APPROVED** [review]

File: src/events.js (hunk 6 above)
Comment (we-ai) [reply src/events.js]: Removed.

**Review by anthonypetersen — APPROVED** [review]
awesome

**Pull request merged**